### PR TITLE
Add more tests to bsp4j and sync with bloop

### PR DESCRIPTION
**Pull request by jvican**
This pull request attempts to:

1. Use bsp4s instead of bsp to publish the bsp Scala implementation.
2. Implement a test for `buildTarget/dependencySources`, implemented by
   Bloop in 1.0.0.
3. Implement a test for `buildTarget/test`, recently implemented by
   bloop in the upgraded version.
4. Implement a test for `buildTarget/run`, recently implemented by
   bloop in the upgraded version.

The pull request makes some changes in the bsp4j and bsp4s to be
compliant with the specification. The changes in bsp4s can be now
fetched from our Scala Center bintray repository in
https://bintray.com/scalacenter/releases/bsp4s and 
https://bintray.com/scalacenter/releases/bsp4j.

The CI is going to fail since the bloop version this PR upgrades to is
not out yet. It will be soon.

### CODE REVIEW

File: bsp4j/src/main/java/ch/epfl/scala/bsp4j/Protocol.xtend [hunk 1]
```diff
@@ -339,10 +339,10 @@ class DependencySourcesResult {
 }
 @JsonRpcData
 class DependencySourcesItem {
-  @NonNull List<BuildTargetIdentifier> targets
```
Comment (olafurpg) [inline bsp4j/src/main/java/ch/epfl/scala/bsp4j/Protocol.xtend]: Good catch! 👍 Sloppy copy-pasting on my half 😅

**Comment by jvican** [comment]
When GitHub is back from vacation, it should run the CI for the latest commit that upgrades to the latest bloop. Then, all tests should pass.

File: build.sbt [hunk 2]
```diff
@@ -68,7 +81,7 @@ lazy val bsp4j = project
       "org.eclipse.lsp4j" % "org.eclipse.lsp4j.generator" % "0.5.0",
       "org.eclipse.lsp4j" % "org.eclipse.lsp4j.jsonrpc" % "0.5.0",
       "org.scala-sbt.ipcsocket" % "ipcsocket" % "1.0.0" % Test,
-      "ch.epfl.scala" % "bloop-frontend_2.12" % "1.0.0" % Test,
+      "ch.epfl.scala" % "bloop-frontend_2.12" % "46e63fc3" % Test,
```
Comment (jvican) [inline build.sbt]: I'm investigating the reason why the CI is publishing these versions without the 1.0.0 prefix. As it's not critical, I propose to merge the PR with this version.

**Comment by olafurpg** [comment]
Great job @jvican ! I have already been using this PR successfully in metals 👍

**Pull request merged**